watchtower/wtwire/session_init.go

```diff
@@ -0,0 +1,78 @@
+package wtwire
+
+import (
+	"io"
+
+	"github.com/lightningnetwork/lnd/lnwallet"
+)
+
+// SessionInit is sent from a client to tower when to negotiate a session, which
+// specifies the total number of updates that can be made, as well as fee rates.
+// An update is consumed by uploading an encrypted blob that contains
+// information required to sweep a revoked commitment transaction.
+type SessionInit struct {
+	// Version specifies the blob format that must be used by all updates
+	// sent under the session key used to negotiate this session.
+	Version uint16
+
+	// MaxUpdates is the maximum number of updates the watchtower will honor
+	// for this session.
+	MaxUpdates uint16
```

Comment:
Any particular reason why this is constraint to a `uint16`? 65k-ish updates _is_ quite a bit, but one could imagine the private watch towers would want to simplify their client code a bit, allowing one session to be re-used for the entire duration. Or perhaps there's some underlying incentive to promote sessions with a finite lifetime?

Reply:
There's no absolute reason to cap at 65k, tho 256 is definitely too small. Ultimately the real limit can always be clamped based on the policy of the tower (public or private). Some thoughts on why we should prefer smaller sessions:

A session constitutes a promise on the watchtower's behalf to reserve _future_ space for the client. The primary concern here is key loss, as the client won't be able to access this state, and the tower can't provably verify such an event. Allowing large sessions means that towers will be more likely to end up with space for which they've committed, but which will go unused. Smaller, more granular sessions with semi-frequent renegotiation would prevent this for the most part.

There is also an economic benefit to clients favoring small sessions in the key loss scenario, as it limits the number of slots a client can lose access to. Since (in the future) those updates will be paid for upfront, this also limits the total amount of funds a client might effectively lose in the process.

Lastly, I think there are decently strong reasons for preferring more ephemeral sessions when it comes to privacy. If one session can cover all (or at least a large number) of updates, there will be little incentive to actually implement session renegotiation and/or rotation, leading to fractured anonymity/behavioral sets.